### PR TITLE
`z_obj_tokei_tobira` OK undocumented

### PR DESCRIPTION
**Pull request by AbdooOwd**
# `z_obj_tokei_tobira`
* OK (Matching)
* Undocumented

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 1]
```diff
@@ -29,23 +30,140 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
 };
 
 // static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80ABD750[] = {
+static InitChainEntry sInitChain[] = {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:33]: You can remove the comment here

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 2]
```diff
@@ -29,23 +30,140 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
+        CutsceneManager_Queue(this->dyna.actor.csId);
+    }
+}
+
+void ObjTokeiTobira_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    ObjTokeiTobira* this = THIS;
+    Player* player = GET_PLAYER(play);
+    s32 pad50;
+    s32 temp_param = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:98]: You can rename this as `type`.
```suggestion
    s32 type = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
```
Don't forget to rename all the uses

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 3]
```diff
@@ -29,23 +30,140 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
+                    this->unk168 = 1;
+                    if (this->unk16C <= 0) {
+                        Actor_PlaySfx(&this->dyna.actor, NA_SE_OC_WOOD_GATE_OPEN);
+                        this->unk16C = 0x50;
+                    }
+                }
+
+                if (sp40 > 48.0f) {
+                    ObjTokeiTobira* tobira;
+                    if (temp_param == OBJTOKEITOBIRA_TYPE_0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:122]: Leave an empty line between the variable declaration and the rest of the code
```suggestion
                    ObjTokeiTobira* tobira;

                    if (temp_param == OBJTOKEITOBIRA_TYPE_0) {
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h [hunk 4]
```diff
@@ -1,17 +1,29 @@
...
-    /* 0x144 */ char unk_144[0x18];
+    /* 0x000 */ DynaPolyActor dyna;
     /* 0x15C */ ObjTokeiTobiraActionFunc actionFunc;
-    /* 0x160 */ char unk_160[0x10];
-} ObjTokeiTobira; // size = 0x170
+    /* 0x160 */ f32 unk160;
+    /* 0x164 */ f32 unk164;
+    /* 0x168 */ s32 unk168;
+    /* 0x16C */ s32 unk16C;
+} ObjTokeiTobira;    /* size = 0x170 */
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:27]: We usually use this type of comment for the size of the structs
```suggestion
} ObjTokeiTobira; // size = 0x170
```

**Comment by AbdooOwd** [comment]
Only needs to run `./format.p`

**Review by AngheloAlf — APPROVED** [review]

**Comment by AngheloAlf** [comment]
Great work! PR looks good.

I'd like to mention that we are trying to get everybody who has contributed and/or wants to contribute to MM decomp to agree to licensing the repo before merging PRs from people who has not yet agreed to it. It is better explained here: https://github.com/louist103/mm-licensing/issues/1.

Could you please read about it and post in that issue if you agree or not. Thanks!

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 5]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+Vec3f D_80ABD760 = { 0.0f, 0.0f, 80.0f };
+s16 D_80ABD76C[] = { 0xC000, 0x4000 };
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:40]:
```suggestion
s16 D_80ABD76C[] = { -0x4000, 0x4000 };
```
0xC000 = -0x4000 as an `s16`

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 6]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/func_80ABD3B0.s")
+    Matrix_RotateYS(D_80ABD76C[type] + this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_MultVec3f(&D_80ABD760, &pos);
+    this->dyna.actor.world.pos.x += pos.x;
+    this->dyna.actor.world.pos.y += pos.y;
+    this->dyna.actor.world.pos.z += pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Update.s")
+    if ((type == OBJTOKEITOBIRA_TYPE_0) && !(CHECK_WEEKEVENTREG(WEEKEVENTREG_59_04)) &&
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:68]:
```suggestion
    if ((type == OBJTOKEITOBIRA_TYPE_0) && !CHECK_WEEKEVENTREG(WEEKEVENTREG_59_04) &&
```
Don't need the extra parens

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 7]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
+        this->unk16C -= 1;
+    }
+
+    this->unk168 = 0;
+    this->unk160 *= 0.87f;
+    this->unk164 += this->unk160;
+    if (this->unk164 * sp48 > 14336.0f) {
+        this->unk160 *= 0.1f;
+        this->unk164 = 14336.0f * sp48;
+    } else if (this->unk164 * sp48 < -1000.0f) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:155]:
```suggestion
    if ((this->unk164 * sp48) > 14336.0f) {
        this->unk160 *= 0.1f;
        this->unk164 = 14336.0f * sp48;
    } else if ((this->unk164 * sp48) < -1000.0f) {
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 8]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
+        Actor_OffsetOfPointInActorCoords(&this->dyna.actor, &sp24, &player->actor.world.pos);
+
+        if ((sp24.z > 0.0f) && (sp24.z < 30.0f)) {
+            this->unk168 = 1;
+            this->unk160 += sp48 * 290.0f;
+        }
+    }
+
+    if (this->unk16C > 0) {
+        this->unk16C -= 1;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:146]:
```suggestion
        this->unk16C--;
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 9]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
+            Actor_OffsetOfPointInActorCoords(&this->dyna.actor, &sp34, &player->actor.world.pos);
+            sp40 = sp34.x * sp48;
+
+            if (sp40 > 20.0f) {
+                if (sp34.z > 0.0f) {
+                    this->unk160 += (sp40 - 20.0f) * sp48 * (16.0f / 3.0f);
+                    this->unk168 = 1;
+                    if (this->unk16C <= 0) {
+                        Actor_PlaySfx(&this->dyna.actor, NA_SE_OC_WOOD_GATE_OPEN);
+                        this->unk16C = 0x50;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:115]:
```suggestion
                        this->unk16C = 80;
```
Looks like some kind of timer so is better as decimal

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h [hunk 10]
```diff
@@ -1,17 +1,29 @@
 #ifndef Z_OBJ_TOKEI_TOBIRA_H
 #define Z_OBJ_TOKEI_TOBIRA_H
 
+#define OBJTOKEITOBIRA_GET_TYPE(thisx) ((thisx)->params & 1)
+#define OBJTOKEITOBIRA_PARAM(type) (type)
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:5]:
```suggestion
#define OBJTOKEITOBIRA_PARAMS(type) (type)
```
Kind of nitpicky, but plural seems better for these bit packing params macros.

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 11]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/func_80ABD3B0.s")
+    Matrix_RotateYS(D_80ABD76C[type] + this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_MultVec3f(&D_80ABD760, &pos);
+    this->dyna.actor.world.pos.x += pos.x;
+    this->dyna.actor.world.pos.y += pos.y;
+    this->dyna.actor.world.pos.z += pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Update.s")
+    if ((type == OBJTOKEITOBIRA_TYPE_0) && !(CHECK_WEEKEVENTREG(WEEKEVENTREG_59_04)) &&
+        (play->sceneId == SCENE_CLOCKTOWER) && (gSaveContext.sceneLayer == 0) && (this->dyna.actor.csId >= 0)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:69]:
```suggestion
        (play->sceneId == SCENE_CLOCKTOWER) && (gSaveContext.sceneLayer == 0) && (this->dyna.actor.csId > CS_ID_NONE)) {
```
Sorry missed this the first time.

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 12]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
     ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+Vec3f D_80ABD760 = { 0.0f, 0.0f, 80.0f };
+s16 D_80ABD76C[] = { -0x4000, 0x4000 };
+CollisionHeader* D_80ABD770[] = { &object_tokei_tobira_Colheader_0012B0, &object_tokei_tobira_Colheader_001590 };
+f32 D_80ABD778[] = { 1.0f, -1.0f };
+Gfx* D_80ABD780[] = { object_tokei_tobira_DL_001108, object_tokei_tobira_DL_0013E8 };
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:43]:
```suggestion
s16 D_80ABD76C[] = {
    -0x4000, // OBJTOKEITOBIRA_TYPE_0
    0x4000,  // OBJTOKEITOBIRA_TYPE_1
};
CollisionHeader* D_80ABD770[] = {
    &object_tokei_tobira_Colheader_0012B0, // OBJTOKEITOBIRA_TYPE_0
    &object_tokei_tobira_Colheader_001590, // OBJTOKEITOBIRA_TYPE_1
};
f32 D_80ABD778[] = {
    1.0f,  // OBJTOKEITOBIRA_TYPE_0
    -1.0f, // OBJTOKEITOBIRA_TYPE_1
};
Gfx* D_80ABD780[] = {
    object_tokei_tobira_DL_001108, // OBJTOKEITOBIRA_TYPE_0
    object_tokei_tobira_DL_0013E8, // OBJTOKEITOBIRA_TYPE_1
};
```
All this data is indexed by type, so it's helpful to put comments on the data to visualize which type maps to which array element

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 13]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
+                           this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z,
+                           OBJTOKEITOBIRA_PARAMS(OBJTOKEITOBIRA_TYPE_1));
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/func_80ABD3B0.s")
+    Matrix_RotateYS(D_80ABD76C[type] + this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_MultVec3f(&D_80ABD760, &pos);
+    this->dyna.actor.world.pos.x += pos.x;
+    this->dyna.actor.world.pos.y += pos.y;
+    this->dyna.actor.world.pos.z += pos.z;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:66]:
```suggestion
    this->dyna.actor.world.pos.x += posOffset.x;
    this->dyna.actor.world.pos.y += posOffset.y;
    this->dyna.actor.world.pos.z += posOffset.z;
```
Position isn't really correct here. It's more an offset from the base position

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 14]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
+    } else {
+        CutsceneManager_Queue(this->dyna.actor.csId);
+    }
+}
+
+void ObjTokeiTobira_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    ObjTokeiTobira* this = THIS;
+    Player* player = GET_PLAYER(play);
+    s32 pad50;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:96]:
```suggestion
    s32 pad1;
    ObjTokeiTobira* this = THIS;
    Player* player = GET_PLAYER(play);
    s32 pad2;
```
Usually if there's more than 1 pad, we name them `pad1` and `pad2`

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c [hunk 15]
```diff
@@ -28,24 +29,141 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
...
+    s32 type = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
+    f32 sp48 = D_80ABD778[type];
+    s32 pad44;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:99]:
```suggestion
    s32 pad3;
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h [hunk 16]
```diff
@@ -1,17 +1,29 @@
 #ifndef Z_OBJ_TOKEI_TOBIRA_H
 #define Z_OBJ_TOKEI_TOBIRA_H
 
+#define OBJTOKEITOBIRA_GET_TYPE(thisx) ((thisx)->params & 1)
+#define OBJTOKEITOBIRA_PARAMS(type) (type)
+
 #include "global.h"
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:7]:
```suggestion
#include "global.h"

#define OBJTOKEITOBIRA_GET_TYPE(thisx) ((thisx)->params & 1)
#define OBJTOKEITOBIRA_PARAMS(type) (type)
```
We typically have includes at the top of the file, so the params can be moved down

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h [hunk 17]
```diff
@@ -1,17 +1,29 @@
...
-    /* 0x144 */ char unk_144[0x18];
+    /* 0x000 */ DynaPolyActor dyna;
     /* 0x15C */ ObjTokeiTobiraActionFunc actionFunc;
-    /* 0x160 */ char unk_160[0x10];
-} ObjTokeiTobira; // size = 0x170
+    /* 0x160 */ f32 unk160;
+    /* 0x164 */ f32 unk164;
+    /* 0x168 */ s32 unk168;
+    /* 0x16C */ s32 unk16C;
+} ObjTokeiTobira;   // size = 0x170
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:27]:
```suggestion
} ObjTokeiTobira; // size = 0x170 
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h [hunk 18]
```diff
@@ -2,16 +2,27 @@
 #define Z_OBJ_TOKEI_TOBIRA_H
 
 #include "global.h"
+#define OBJTOKEITOBIRA_GET_TYPE(thisx) ((thisx)->params & 1)
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:5]:
```suggestion
#include "global.h"

#define OBJTOKEITOBIRA_GET_TYPE(thisx) ((thisx)->params & 1)
```
Small nit but can you add a newline here.

**Review by hensldm — APPROVED** [review]
LGTM. Btw, if someone has already approved the PR there is no need to request another review from them. I'm mentioning this since Anghelo had already approved your PR, but by requesting for a review it no longer shows up.

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h (hunk 18 above)
Comment (engineer124) [reply src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:5]: Bump

Comment (hensldm) [reply src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:5]: Would like to bump on this again and see if you could address this small little nit.

**Review by engineer124 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Comment by AbdooOwd** [comment]
LEZGOOO MERGED!!!11!!1!!!!

**Pull request merged**